kong/conf_loader/init.lua

```diff
@@ -1902,6 +1946,73 @@ local function load(path, custom_conf, opts)
     end
   end
 
+  -- WebAssembly module support
+  if conf.wasm then
```

Comment:
The slack thread I tagged you in has more details, but I think there are a couple notable directives we should dropping, warning about, or just throwing an error if we see that the user has defined them.

I'll defer to your judgment on what we should do about these (if anything)--just bringing 'em to your attention.

* `proxy_wasm_request_headers_in_access` is just irrelevant and will have no effect: warn + drop?
* `wasm_call` is the main nginx.conf (non-FFI) entrypoint for executing wasm modules, so it conflicts with our use case. At best it won't do anything, but at worse it could cause breakage or weird behavior: warn + drop? or error?

Aside from that there are a couple noteworthy directives, mostly around DNS:

* `resolver`/`resolver_add`/`resolver_timeout` should probably be avoided by users, but we don't guard against this elsewhere, so I think these are not worth doing anything about
* `proxy_wasm_lua_resolver` this will be turned on by default and should not be turned off under normal circumstances, but it sounds like we want to allow users to alter this setting if it's really desired. Warn, maybe, or just do nothing and continue on?

https://github.com/Kong/ngx_wasm_module/blob/main/docs/DIRECTIVES.md

Reply:
Thanks for the context. I've carded KAG-2087 for this. Will open a separate PR.